mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -214,7 +216,7 @@ private[ml] object TreeClassifierParams {
 /**
  * Parameters for Decision Tree-based regression algorithms.
  */
-private[ml] trait TreeRegressorParams extends Params {
+private[ml] trait TreeRegressorParams extends DecisionTreeParams with HasVarianceCol {
```

Comment:
This patch does not add varianceCol to all tree and ensemble regressors.  HasVarianceCol should only be added to DecisionTreeRegressor.